opm/simulators/wells/WellInterface_impl.hpp

```diff
@@ -1013,14 +1013,22 @@ namespace Opm
                 }
             }
         }
+
+        // Calculate the rates that follow from the current primary variables.
+        std::vector<double> well_q_s = computeCurrentWellRates(ebosSimulator, deferred_logger);
+
         if (nonzero_rate_index == -1) {
             // No nonzero rates.
+            // Use the computed rate directly
+            const int comp_idx_nz = this->flowPhaseToEbosCompIdx(nonzero_rate_index);
+            for (int p = 0; p < this->number_of_phases_; ++p) {
+                const int comp_idx = this->flowPhaseToEbosCompIdx(p);
+                double& rate = ws.surface_rates[p];
+                rate = well_q_s[comp_idx];
+            }
```

Comment:
The `comp_idx_nz` appears not to be used anywhere and the rest of the loop **could** be condensed to
```C++
for (int p = 0; p < this->number_of_phases; ++p) {
    ws.surface_rates[p] = well_q_s[this->flowPhaseToEbosCompIdx(p)];
}
```
if we wanted to.  I'm not suggesting that we write it like the above, I just want to make sure that this is the intended semantics.